.github/workflows/canary-integration-test.yml

```diff
@@ -91,11 +91,18 @@ jobs:
         name: canary
         path: test
 
-    - name: setup tmate session for debugging
-      if: failure()
+    - name: setup tmate session for debugging when event is PR
+      if: failure() && github.event_name == 'pull_request'
```

Comment:
How about we keep this change as simple as possible? I'm not sure it's worth doing all these checks just for the tmate session. I think it would be fine to have a consistent timeout for each workflow. For example, the integration-tests-on-release.yml workflow can have a shorter timeout of 30 since it's never running PRs. Since the canary workflow is running PRs as well, it's fine to have a longer timeout since it doesn't have as big a test matrix. But do we really need 120 minutes? Maybe 60 is fine?

Reply:
okay, changed to 60m